Fix PackageManager Remove button not working, add unit-tests & CI

VetleNeumann opened the pull request:
PackageManager Remove button stopped working in Unity-6000.*, due to Unity changing how `OnSelectionChanged` worked, making it so `RemoveSourcesClicked` found zero selected sources.
To reduce complexity in making the change work with Unity-2018, made the `Package Source Settings` list be single-select, instead of multiselect. If this is unacceptable I can rework it again.

Added unit-tests for the Add/Remove buttons on the `Package Source Settings`. To make some of the operations/logic accessible for the unit-test file, split up some functions to isolated functions and made some operations public, like `GetAvailablePackageSourceTypes()` or `RemoveSource()`.
The unit-tests are added using Unity's unit-test system, which uses NUnit. This lets you use the Unity `Test Runner` (opened with `Window->General->Test Runner`), to run tests:
<img width="725" height="485" alt="image" src="https://github.com/user-attachments/assets/1a3149db-1a9c-495d-84aa-e9f0bbd6d846" />

Updated the .gitignore, using same source as prior (https://www.gitignore.io/?templates=unity).

Added CI, which builds a small test-project which includes ThunderKit as a package, as well as running the unit-tests through it. Currently, builds and runs on these Unity versions:
- **6000.0.53f1**
- **2022.1.24f1**
- **2020.3.48f1**
- **2019.4.40f1**
- **2018.4.36f1**
The pipeline uses Game-CI actions, which requires [some setup of GitHub Secrets](https://game.ci/docs/github/activation) on the project to run. [Check it out](https://github.com/VetleNeumann/ThunderKit/actions/runs/16425364406):
<img width="1246" height="879" alt="image" src="https://github.com/user-attachments/assets/9fedb433-f2eb-4531-9f05-4846088aabd9" />

PassivePicasso commented:
This looks superb overall, the changes are more than I would have done, but serve to make the code cleaner than I would have left it. Lets just change those few things and then I'll merge and push tag 9.1.3

VetleNeumann commented:
@PassivePicasso Added test fixtures to make them a bit cleaner, and added some unit-tests to test correct package sources are showing up in the Package Manager window as well. 👍Fixed the bug where removing a source, while having the Package Manager window open, caused an error (Package Manager trying to access the now removed source).

Added GitHub Actions caching on Unity files, as recommended by game-ci.

PR should be ready from my end now, really happy with it.
